doc/qos/tunnel_dscp_remapping.md

```diff
@@ -0,0 +1,202 @@
+# DSCP remapping for tunnel traffic
+
+## 1 Table of Content ###
+
+- [Revision](#11-revision)
+- [Scope](#2-scope)
+- [Definitions/Abbreviations](#3-definitionsabbreviations)
+- [Overview](#4-overview)
+- [Design](#5-design)
+    - [SWSS Schema](#51-swss-schema)
+        - [Define new table for mapping](#511-define-new-table-for-mapping)
+        - [Update existing TUNNEL table](#512-update-existing-tunnel-table)
+        - [Define new field for extra lossless queues](#513-define-new-field-for-extra-lossless-queues)
+    - [SAI attribute](#52-sai-attribute)
+    - [orchagent](#53-orchagent)
+- [Test requirement](#6-test-requirement)
+- [Open Questions](#7-open-questions)
+
+### 1.1 Revision ###
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Bing Wang   | Initial version                   |
+
+
+## 2 Scope ##
+
+This document covers high level design of DSCP and TC remapping for tunnel traffic in SONiC.
+
+## 3 Definitions/Abbreviations ##
+
+
+| Term | Meaning |
+|:--------:|:---------------------------------------------:|
+| PFC | Priority-based Flow Control  |
+| TC | Traffic class|
+| DSCP| Differentiated Services Code Point |
+
+## 4 Overview
+
+The current QoS map architecture allows for port-based selection of each QoS map. However, we are not able to override the port-based QoS map for tunnel traffic. 
+This design proposes a method to remapping DSCP and TC for IPinIP tunnel. 
+
+
+## 5 Design ##
+
+### 5.1 SWSS Schema
+#### 5.1.1 Define new table for mapping
+Please be noted that below config is to remap traffic in queue 3 to queue 2, and traffic in queue 4 to queue 6.
+* Table for decap
+
+    DSCP_TO_TC_MAP for mapping DSCP to TC
+    ```json
+    "DSCP_TO_TC_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "1",
+            "1": "1",
+            "2": "2",
+            "3": "3",
+            "4": "4",
+            "5": "1",
+            "6": "6",
+            "7": "1",
+            "8": "0",
+            "9": "1"
+    }
+    ```
+    TC_TO_PRIORITY_GROUP_MAP for mappping TC to PG
+    ```json
+    "TC_TO_PRIORITY_GROUP_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "0",
+            "1": "0",
+            "2": "0",
+            "3": "2",
+            "4": "6",
+            "5": "0",
+            "6": "0",
+            "7": "7"
+    }
+    ```
+* Table for encap
+
+    TC_TO_QUEUE_MAP for remapping queue
+    ```json
+    "TC_TO_QUEUE_MAP": {
+        "AZURE_TUNNEL": {
+            "0": "0",
+            "1": "1",
+            "2": "1",
+            "3": "2",
+            "4": "6",
+            "5": "5",
+            "6": "1",
+            "7": "7"
+    }
+    ``` 
+    TC_TO_DSCP_MAP for rewriting DSCP
+    ```json
+    "TC_TO_DSCP_MAP": {
```

Comment:
Since this is a new table, we will need to define yang model for this

Reply:
Updated. Thanks